### ISSUE TITLE
Is it possible to have the bot change nicknames rather than delete messages. It makes matrix and IRC bridges unuseable.

### ISSUE THREAD

**Issue by ameliafloof** [xSke/PluralKit#358]
Pluralkit is a great bot but the way it deletes messages does not integrate well with platforms like matrix (where deletions are typically logged) and IRC where messages can not be redacted,

I run a FOSS community (where matrix and IRC bridges are needed for some people) and it spams the bridges and makes chat unreadable from not on discord unless using specific clients in settings that many people may not have.

**Comment by xSke** [comment]
No. If that was possible we'd have been doing it the whole time.

**Comment by ameliafloof** [comment]
Alright thank you for letting me know!

**Comment by xSke** [comment]
To clarify: it's possible for bots to change nicknames of other users, but those nickname changes don't consistently propagate, they don't affect avatars, and they don't "stick" to the message. The current method is about the best we can do while still maintaining the effects. I know this can create issues with other things, but there's not much to do about that other than live with it or turn the bot off.

**Comment by Qyriad** [comment]
A not-great but usable workaround for the bridge problem is to have the bridge ignore accounts with a certain role, and then have those accounts speak *only* with PluralKit.

**Comment by spiralw** [comment]
I'm currently writing (well, planning to write) a Matrix/IRC bridge that allows for PluralKit on the Discord side. We'll make this open-source when we get around to making it, of course, but the idea behind it is to wait a second and then ask the [PluralKit API](https://pluralkit.me/api/) if the message was proxied, and if so don't send it over the bridge. The delay is annoying, but it's an option, I guess.
